common/Scripts/helpers.sh

```diff
@@ -1,30 +1,49 @@
 #!/usr/bin/env bash
 
+# shellcheck wants to add '|| exit 1' willy nilly, so let's
+# build a nice 'failmsg' function instead.
+function failmsg() {
+    echo "$*"
+    return 1
+}
+
 # Goes to the root directory of the solus packages
 # git repository from anywhere on the filesystem.
 # This function will only work if this script is sourced
 # by your bash shell.
 function gotosoluspkgs() {
-    cd "$(dirname "$(readlink "${BASH_SOURCE[0]}")")/../../" || exit 1
+    cd "$(dirname "$(readlink "${BASH_SOURCE[0]}")")/../../" || \
+        failmsg "Unable to find the solus packages git dir."
 }
 
 # Goes to the root directory of the git repository
 function goroot() {
-    cd "$(git rev-parse --show-toplevel)" || exit 1
+    local pkgroot="$(git rev-parse --show-toplevel)"
+    [[ -n "${pkgroot}" ]] && cd "${pkgroot}" || \
+        failmsg "If you're not somewhere in the solus packages git dir, use gotosoluspkgs instead."
 }
 
 # Push into a package directory
 function gotopkg() {
-    cd "$(git rev-parse --show-toplevel)"/packages/*/"$1" || exit 1
+    local arg="$1"
+    gotosoluspkgs
+    if [[ -n "$arg" ]]; then
+        cd "$(git rev-parse --show-toplevel)"/packages/*/"$arg" || \
+            failmsg "Something went wrong."
+    else
+        failmsg "No package specified, going to the root of the solus packages git dir instead."
+    fi
 }
 
 # What provides a lib
 function whatprovides() {
+    gotosoluspkgs
```

Comment:
I agree with @sheepman4267 that forcing the shell into a certain directory when it didn't before is disruptive to the user. This is not necessary if you use [git -C](https://git-scm.com/docs/git).

Generally, if I need code to figure something out, I try to do it in a way that doesn't affect ther user environment if I can.